patchy/api.py

```diff
@@ -148,8 +148,38 @@ def _get_source(func):
         return source
 
 
-def _set_source(func, new_source):
+def _class_name(func):
+    qualname = getattr(func, '__qualname__', None)
+    if qualname is not None:
+        split_name = qualname.split('.')
+        try:
+            class_name = split_name[-2]
+        except IndexError:
+            return None
+        else:
+            if class_name == '<locals>':
+                return None
+            return class_name
+    im_class = getattr(func, 'im_class', None)
+    if im_class is not None:
+        return im_class.__name__
+
+
+def _indent(new_source):
+    return '\n'.join(['    ' + x for x in new_source.split('\n')])
```

Comment:
mess up docstrings only slightly?

Also yes there will be a `TabError`, you could add a test for that